### PR TITLE
feat(connector): [Multisafepay] add error handling

### PR DESCRIPTION
**Pull request by AkshayaFoiger**
## Type of Change

- [ ] Bugfix
- [x] New feature
- [ ] Enhancement
- [ ] Refactoring
- [ ] Dependency updates
- [ ] Documentation
- [ ] CI/CD

### Description 
- Resolves [#2636](https://github.com/juspay/hyperswitch-cloud/issues/2636)

### Additional Changes

- [ ] This PR modifies the API contract
- [ ] This PR modifies the database schema
- [ ] This PR modifies application configuration/environment variables

resolves 

## Checklist

- [x] I formatted the code `cargo +nightly fmt --all`
- [x] I addressed lints thrown by `cargo clippy`
- [x] I reviewed the submitted code
- [ ] I added unit tests for my changes where possible
- [ ] I added a [CHANGELOG](/CHANGELOG.md) entry if applicable

### CODE REVIEW

File: crates/router/src/connector/multisafepay/transformers.rs [hunk 1]
```diff
@@ -605,61 +624,93 @@ pub struct RefundData {
...
+                        refund_status: enums::RefundStatus::from(refund_stat),
+                    }),
+                    ..item.data
+                })
+            }
+            MultisafepayRefundResponse::ErrorResponse(error_response) => Ok(Self {
+                response: Err(types::ErrorResponse {
+                    code: error_response.error_code.to_string(),
+                    message: error_response.error_info,
+                    reason: None,
```
Comment (ArjunKarthik) [inline crates/router/src/connector/multisafepay/transformers.rs:667]: Populate reason with error_response.error_info

File: crates/router/src/connector/multisafepay/transformers.rs [hunk 2]
```diff
@@ -498,57 +498,76 @@ pub struct MultisafepayPaymentsResponse {
...
-                network_txn_id: None,
-                connector_response_reference_id: Some(item.response.data.order_id.clone()),
+                    ..item.data
+                })
+            }
+            MultisafepayAuthResponse::ErrorResponse(error_response) => Ok(Self {
+                response: Err(types::ErrorResponse {
+                    code: error_response.error_code.to_string(),
+                    message: error_response.error_info,
+                    reason: None,
```
Comment (ArjunKarthik) [inline crates/router/src/connector/multisafepay/transformers.rs:565]: Populate reason with error_response.error_info

File: crates/router/src/connector/multisafepay/transformers.rs [hunk 3]
```diff
@@ -605,61 +624,93 @@ pub struct RefundData {
...
+                        refund_status: enums::RefundStatus::from(refund_status),
+                    }),
+                    ..item.data
+                })
+            }
+            MultisafepayRefundResponse::ErrorResponse(error_response) => Ok(Self {
+                response: Err(types::ErrorResponse {
+                    code: error_response.error_code.to_string(),
+                    message: error_response.error_info,
+                    reason: None,
```
Comment (ArjunKarthik) [inline crates/router/src/connector/multisafepay/transformers.rs:703]: same here

**Review by ArjunKarthik — CHANGES_REQUESTED** [review]

**Review by ArjunKarthik — APPROVED** [review]

File: crates/router/src/connector/multisafepay/transformers.rs [hunk 4]
```diff
@@ -605,61 +624,93 @@ pub struct RefundData {
...
-            RefundStatus::Failed
-        };
-
-        Ok(Self {
-            response: Ok(types::RefundsResponseData {
-                connector_refund_id: item.response.data.refund_id.to_string(),
-                refund_status: enums::RefundStatus::from(refund_stat),
+        match item.response {
+            MultisafepayRefundResponse::RefundResponse(refund_data) => {
+                let refund_stat = if refund_data.success {
```
Comment (prasunna09) [inline crates/router/src/connector/multisafepay/transformers.rs:649]:
```suggestion
                let refund_status = if refund_data.success {
```

File: crates/router/src/connector/multisafepay/transformers.rs [hunk 5]
```diff
@@ -605,61 +624,93 @@ pub struct RefundData {
...
+                    RefundStatus::Succeeded
+                } else {
+                    RefundStatus::Failed
+                };
+
+                Ok(Self {
+                    response: Ok(types::RefundsResponseData {
+                        connector_refund_id: refund_data.data.refund_id.to_string(),
+                        refund_status: enums::RefundStatus::from(refund_stat),
```
Comment (prasunna09) [inline crates/router/src/connector/multisafepay/transformers.rs:658]:
```suggestion
                        refund_status: enums::RefundStatus::from(refund_status),
```

File: crates/router/src/connector/multisafepay/transformers.rs [hunk 6]
```diff
@@ -498,57 +498,76 @@ pub struct MultisafepayPaymentsResponse {
...
+                    .clone()
+                    .map(|url| services::RedirectForm::from((url, services::Method::Get)));
+
+                let default_status = if payment_response.success {
+                    MultisafepayPaymentStatus::Initialized
+                } else {
+                    MultisafepayPaymentStatus::Declined
+                };
+
+                let status = payment_response.data.status.unwrap_or(default_status);
```
Comment (prasunna09) [inline crates/router/src/connector/multisafepay/transformers.rs:535]: why are we using `unwrap_or` here?

**Review by prasunna09 — CHANGES_REQUESTED** [review]

**Comment by prasunna09** [comment]
@AkshayaFoiger Please take these changes in separate PR.

**Review by prasunna09 — APPROVED** [review]

**Review by ArjunKarthik — DISMISSED** [review]

**Review by prasunna09 — DISMISSED** [review]

**Comment by AkshayaFoiger** [comment]
Test Case
1. Create a payment with incorrect Multisafepay API key. This should populate appropriate error code and error message in the response.

**Comment by Gnanasundari24** [comment]
@AkshayaFoiger Kindly update the postman collections for the supported connectors and update the description with the relevant screenshots.
You can refer to the rustman documentation [here](https://github.com/juspay/hyperswitch/blob/main/crates/test_utils/README.md)

**Review by ArjunKarthik — APPROVED** [review]

**Review by prasunna09 — APPROVED** [review]

**Review by Gnanasundari24 — APPROVED** [review]

**Pull request merged**